File: akka-docs/src/test/java/jdocs/stream/operators/converters/StreamConvertersToJava.java
/*
 * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
 */

package jdocs.stream.operators.converters;

import akka.NotUsed;
import akka.actor.ActorSystem;
// #import
import akka.stream.Materializer;
import akka.stream.javadsl.Sink;
import akka.stream.javadsl.StreamConverters;
import java.util.stream.Stream;
// #import
import akka.testkit.javadsl.TestKit;
import jdocs.AbstractJavaTest;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import akka.stream.javadsl.Source;

import static org.junit.Assert.assertEquals;

/** */
public class StreamConvertersToJava extends AbstractJavaTest {

  static ActorSystem system;
  static Materializer mat;

  @BeforeClass
  public static void setup() {
    system = ActorSystem.create("StreamComvertersToJava");
    mat = Materializer.matFromSystem(system);
  }

  @AfterClass
  public static void tearDown() {
    TestKit.shutdownActorSystem(system);
    system = null;
  }

  @Test
  public void demonstrateConverterToJava8Stream() {
    // #asJavaStream

    Source<Integer, NotUsed> source = Source.range(0, 9).filter(i -> i % 2 == 0);

    Sink<Integer, Stream<Integer>> sink = StreamConverters.<Integer>asJavaStream();

    Stream<Integer> jStream = source.runWith(sink, mat);
    assertEquals(5, jStream.count());

    // #asJavaStream
  }
}

File: akka-docs/src/test/scala/docs/stream/operators/converters/StreamConvertersToJava.scala
/*
 * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.stream.operators.converters

import java.util.stream.Collectors

import akka.NotUsed
import akka.stream.scaladsl.Keep
import akka.stream.scaladsl.RunnableGraph
import akka.stream.scaladsl.Source
import akka.stream.scaladsl.StreamConverters
import akka.testkit.AkkaSpec


class StreamConvertersToJava extends AkkaSpec {

  "demonstrate to Java8 streams" in {
    val source: Source[Int, NotUsed] = Source(0 to 9).filter(_ % 2 == 0)
    val akkaStream: RunnableGraph[java.util.stream.Stream[Int]] =
      source.toMat(StreamConverters.asJavaStream())(Keep.right)
    val jStream: java.util.stream.Stream[Int] = akkaStream.run()
    val result:java.lang.Long = jStream.collect(Collectors.counting())

    result should be(5)
  }

}
